app/models/person.rb

```diff
@@ -155,6 +155,18 @@ def self.search(query, user)
     ).includes(:profile).order(search_order)
   end
 
+  def self.contacts_search(query, user)
+    return where("1 = 0") if query.to_s.blank? || query.to_s.length < 2
+
+    sql, tokens = search_query_string(query)
+
+    Person.joins(:contacts)
+          .where(contacts: {user_id: user.id})
+          .where(sql, *tokens)
+          .includes(:profile)
+          .order(search_order)
```

Comment:
So the only difference here to `search` is that it doesn't include people with `searchable: true`. I feel like this could be less redundant, perhaps just a keyword argument to `search`?

```rb
def self.search(query, user, only_contacts: false)
# ...
  query = Person
  query = query.searchable(user) unless only_contacts
  #....